app/commands/update_current_month_datasource_expenses.py

```diff
@@ -93,13 +93,14 @@ async def store_datasource_expenses(
     for organization_id, datasources in datasources_per_organization_id.items():
         for datasource in datasources:
             existing_datasource_expense, created = await datasource_expense_handler.get_or_create(
-                datasource_id=datasource["id"],
+                linked_datasource_id=datasource["id"],
```

Comment:
as a filter for get you should use:

```suggestion
                datasource_id=datasource["account_id"],
```